pkg/grafana/client/client.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/go-logr/logr"
+	grafana "github.com/grafana/grafana-openapi-client-go/client"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	GrafanaUrl           = "http://grafana.monitoring.svc.cluster.local:3000"
+	Namespace            = "monitoring"
+	GrafanaSecretName    = "grafana"
+	grafanaTlsSecretName = "grafana-tls" // nolint:gosec
+)
+
+func GenerateGrafanaClient(ctx context.Context, client client.Client, logger logr.Logger) (*grafana.GrafanaHTTPAPI, error) {
+	// Get grafana admin-password and admin-user
+	adminCredentials, err := getAdminCredentials(ctx, client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch grafana admin secret: %w", err)
```

Comment:
I'm still not sure about the fmt.Error as I quite lack the errors.WithStack that we use all around

Reply:
I agree, I would rather use the same logger as in the reconciler